app/src/js/utils/table-config/granules.js

```diff
@@ -50,41 +49,61 @@ export const tableColumns = [
     Header: 'Status',
     accessor: 'status',
     width: 110,
-    Cell: ({ cell: { value } }) => <Link to={(location) => ({ pathname: `/granules/${value}`, search: getPersistentQueryParams(location) })} className={`granule__status granule__status--${value}`}>{displayCase(value)}</Link> // eslint-disable-line react/prop-types
+    isLink: true,
+    linkTo: (row) => `/granules/${row.status}`,
+    Cell: ({ cell: { value } }) => (<span className={`granule__status granule__status--${value}`}>{displayCase(value)}</span>) // eslint-disable-line react/prop-types
   },
   {
     Header: 'Name',
     accessor: 'granuleId',
-    // eslint-disable-next-line react/prop-types
-    Cell: ({ cell: { value } }) => <CopyCellPopover cellContent={granuleLink(value)} id={`granuleId-${value}-popover`} popoverContent={granuleLink(value)} value={value} />,
-    width: 225
+    width: 225,
+    isLink: true,
+    useCopyCellPopover: true,
+    linkTo: (row) => `/granules/granule/${row.granuleId || row.values?.granuleId || row.original?.granuleId}`
   },
   {
     Header: 'Published',
     accessor: 'published',
+    isLink: true,
     Cell: ({ row: { original: { cmrLink, published } } }) => (// eslint-disable-line react/prop-types
       cmrLink && bool(published) ? <a href={cmrLink} target='_blank'>{bool(published)}</a> : bool(published)
     )
   },
   {
     Header: strings.collection_id,
     accessor: 'collectionId',
-    // eslint-disable-next-line react/prop-types
-    Cell: ({ cell: { value } }) => <CopyCellPopover cellContent={collectionLink(value)} id={`collectionId-${value}-popover`} popoverContent={collectionLink(value)} value={value} />,
+    isLink: true,
+    useCopyCellPopover: true,
+    linkTo: (row) => {
+      const { name, version } = deconstructCollectionId(row.collectionId);
+      return `/collections/collection/${name}/${encodeURIComponent(version)}`;
+    },
+    Cell: ({ cell: { value } }) => collectionLink(value)
   },
   {
     Header: 'Executions List',
     accessor: 'granuleId',
-    Cell: ({ row: { original: { collectionId, granuleId } } }) => (// eslint-disable-line react/prop-types
-      <Link to={(location) => ({ pathname: `/executions/executions-list/${encodeURIComponent(collectionId)}/${encodeURIComponent(path.basename(granuleId))}` })}>link</Link>
-    ),
+    isLink: true,
+    linkTo: (row) => {
+      const collectionId = row.collectionId || row.original?.collectionId;
+      const granuleId = row.granuleId || row.original?.granuleId;
+
+      if (collectionId && granuleId) {
```

Comment:
Does this fix issues where collectionId isn't provided?

Reply:
Yes

Reply:
So how does it relate to the React Router upgrade? Is this an additional feature? Is there a ticket for it?

Reply:
This is a catch. When I was converting the link to work with React Router I wanted to make sure that the routes work and if the user runs into issue there's this fallback.